server/src/main/java/keywhiz/service/resources/automation/v2/AutomationAPIsLoggingImpl.java

```diff
@@ -0,0 +1,15 @@
+package keywhiz.service.resources.automation.v2;
+
+import java.util.Arrays;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.aopalliance.intercept.MethodInvocation;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AutomationAPIsLoggingImpl implements MethodInterceptor {
+  private static final Logger logger = LoggerFactory.getLogger(AutomationAPIsLogging.class);
+  public Object invoke(MethodInvocation invocation) throws Throwable {
+    logger.info(String.format("Function Name: %s, Function Arguments: %s",invocation.getMethod().getName(), Arrays.toString(invocation.getArguments())));
+    return invocation.proceed();
+  }
```

Comment:
I love how simple this is! I still think it would be worthwhile to try to use reflection to get the parameter names automatically, either from the compiled code or from the annotations. For the compiled code, It turns out that the names may actually be present, you can use invocation.getMethod().getParameters() to get at the parameters array and then use the methods isNamePresent() and getName() to grab the name. If it is present I would swap out the value in the arguments array with a Map.Entry containing the name and argument value.

Reply:
Ah, forgot one thing: if you use Arrays.deepToString() it will correctly handle nested arrays. I don't know if we actually have any methods with arrays in their signature but better to get the proper behavior in place now before we need it.